### PR TITLE
Feature/upgrade docx4j #183997533

### PR DESCRIPTION
**Pull request by TheDanDLion**
so our master branch is forked from plutex/master, but it seems they abandoned their master branch a while back in favor of a different branching strategy. they instead make their `default` branch whatever their latest version is (so for us the equivalent would be making release/202X.X.X our main branch every release instead of merging to master). I see two ways we can handle this:

1) Also abandon our master branch and make the release branch the default branch (I'll merge this PR into a forked branch of 11_4_8 from plutx/docx4j)
2) Maintain our branching strategy by using master. I can sync master up to 11_4_8, discard whatever commits it has, then the PR should boil down to the five commits i've got listed below.

let me know if you'd rather keep our branching strategy or use theirs for this fork. I think I would personally like to keep our branching strategy, so I'm opening this against our master.

my commits:
https://github.com/LegalSifter/docx4j/commit/aeb0c27bdee04be5ba995738c3a0a3920cae1a87
https://github.com/LegalSifter/docx4j/commit/848834a7b3a42f7bd1a16f3f9228521a42b7f081
https://github.com/LegalSifter/docx4j/commit/4d087bde772048286bb7a0d72fe1087bc2eef3eb
https://github.com/LegalSifter/docx4j/commit/a3696a2c03a4bcefb1ffc37eab83639d81a500bc
https://github.com/LegalSifter/docx4j/commit/bd0df79b23fbb1afc69c00bff5c000012f149f67

### CODE REVIEW

**Comment by tflavin** [comment]
@TheDanDLion i would prefer master as well. it looks like github is showing several conflicting files, are you able to resolve?
is this ready for review yet? just add me when it is

**Comment by TheDanDLion** [comment]
@tflavin This should be ready to review now, the commits seem to just be mine, thanks for the input

**Comment by TheDanDLion** [comment]
closing & re-opening so the vulnerability scan runs
csharp will fail, but the js & java scans should pass w/ some results

edit: i removed the csharp scan since it's only 0.1% of the repo and ran the scan a second time, the results of the previous scan were a little confusing to interpret

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.git
+.gitignore
```
Comment (tflavin) [inline .dockerignore]: minor: newline

File: pom.xml [hunk 2]
```diff
@@ -26,44 +26,45 @@
...
-
-        <module>docx4j-samples-documents4j-remote</module>        
-        <module>docx4j-samples-documents4j-local</module>        
-        
-        <module>docx4j-bundle</module>        
-        
-        <module>docx4j-legacy-service</module>        
-        
+
+		<!-- commenting out this stuff, it's not needed to build docx4j-core -->
```
Comment (tflavin) [inline pom.xml]: minor: spacing

File: serv/.gitignore [hunk 3]
```diff
@@ -0,0 +1,10 @@
+/target
+/lib
+/classes
+/checkouts
+pom.xml
+pom.xml.asc
+*.jar
+*.class
+/.lein-*
+/.nrepl-port
```
Comment (tflavin) [inline serv/.gitignore:10]: minor: newline

File: serv/project.clj [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+                           [compojure "1.6.1"]
+                           [cheshire "5.7.0"]
+                           [org.slf4j/slf4j-log4j12 "1.7.22"]
+                           [me.raynes/fs "1.4.6"]
+                           [ring/ring-defaults "0.3.2"]]
+            :init-ns serv.handler
+            :plugins [[lein-ring "0.12.5"]]
+            :ring {:handler serv.handler/app}
+            :profiles {:dev {:dependencies [[javax.servlet/servlet-api "2.5"]
+                                            [ring/ring-mock "0.3.2"]]}})
```
Comment (tflavin) [inline serv/project.clj]: minor: newline

File: pom.xml [hunk 5]
```diff
@@ -9,11 +9,11 @@
   </parent>
 
 	<groupId>org.docx4j</groupId>
-	<artifactId>docx4j-parent</artifactId>	
+	<artifactId>docx4j-parent</artifactId>
```
Comment (tflavin) [inline pom.xml:12]: minor/question: usually i appreciate removing whitespace, but does it make more sense to ignore in this kind of situation where we are forking, just so as little as possible is actually changed?

**Review by tflavin — APPROVED** [review]
looks good, tried to test. i see some of the minor comments were on the original, so fix if you want.

FYI i think this is the commit prior to any of the changes - for future reference. https://github.com/LegalSifter/docx4j/tree/b7e8247fa1d600a66614f59d45c8c2186d5f62fb

**Comment by tflavin** [comment]
also i did test a problem doc for each of the fixes and they seem to all still be working fine

File: pom.xml (hunk 5 above)
Comment (TheDanDLion) [reply pom.xml:12]: kind of yeah, at least from a "do as little work as necessary to get it functional" perspective. i'll try to refrain from too much white-space changes but sometimes I can't help myself

**Pull request merged**